### ISSUE TITLE
.at(t) exhibits unusually high CPU usage in Ubuntu

### ISSUE THREAD

**Issue by PeterS-mBryonics** [skyfielders/python-skyfield#916]
This one has me a bit stumped. Running test code on Windows, all is well and good. Running the same code on Linux on two different computers (One WSL, one native, both Ubuntu 22.04) results in unusually high CPU usage across all cores while the code is running. Minimal example below:

```
import time
from skyfield.api import load, wgs84

ts = load.timescale()

bluffton = wgs84.latlon(+40.8939, -83.8917)

t = ts.now()
i = 0
sec = 0
while i < 100:
    t += 1 / 86400
    topocentric = bluffton.at(t)
    time.sleep(0.25)
    i += 1
```

Adjusting the time.sleep affects the overall processor load (removing it maxes out all cores). Each CPU is loaded relatively evenly which is not what I was expecting - as far as I understand it this should be single-threaded. Running two of these programs at once when it's arranged such that one program maxes out the computer does slow them down, so it doesn't appear to be the CPU loading lying to me which was my first thought. Having the slightly weird t+1s bit in the loop is so I can be sure I'm not muddling things by calling ts.now() repeatedly.

I would have expected the CPU usage to go up and down, it only visibly does that at about a sleep time of 0.75 s. I feel like I'm missing something obvious, so I'm happy to be proved a muppet!

**Comment by brandon-rhodes** [comment]
See whether OpenBLAS might be the culprit:

https://rhodesmill.org/skyfield/accuracy-efficiency.html#using-too-many-cpu-cores

Let us know if the fix suggested there affects the behavior you are seeing!

**Comment by PeterS-mBryonics** [comment]
<facepalm> RTFM! Knew I missed something...

A solution for those that are looking, add this to the top of the file, _before_ importing numpy (including skyfield!). An alternative is changing environment variables or a change to .bashrc but I'm 90% sure I'd forget I'd done that and go through the whole thing again.

```
import os
os.environ['OPENBLAS_NUM_THREADS'] = '1'
os.environ['MKL_NUM_THREADS'] = '1'
```

This dealt with the issue

**Comment by brandon-rhodes** [comment]
@PeterS-mBryonics — I have just landed 9c4edda956453b298b24f824ad11db2aa5918472 which updates the Skyfield docs to mention both of these settings, instead of just the first one, and so that the docs also show your maneuver of setting the environment variables right in Python (thanks for the idea!).

I'm glad that the settings worked for you!

**Comment by mikeawalker** [comment]
Confirming that the OPENBLAS_NUM_THREADS environment variable is the root cause of the issue that I'm seeing

Here is a minimal example that triggers the effect:

```
import skyfield.api as sf
import datetime
import time
import pytz
def main(  ):
    id = "ISS (ZARYA)"
    n = 100
    ts = sf.load.timescale( )
    satellites = sf.load.tle_file("https://celestrak.org/NORAD/elements/gp.php?GROUP=stations&FORMAT=tle")
    by_name = {sat.name: sat for sat in satellites}

    sat = by_name[id] 

    start = datetime.datetime.utcnow()
    start = start.replace(tzinfo=pytz.utc)
   
    for k in range( 0 , n ):
        t = start + datetime.timedelta(seconds=k)
        tSF = ts.from_datetime( t )
        state = sat.at( tSF )

        lat,lon = sf.wgs84.latlon_of( state) 
        alt = sf.wgs84.height_of( state ) 
        print( f"{lat},{lon},{alt}")
        time.sleep(1)

if __name__=="__main__":

    main( )
``` 

If i run this example with:

`export OPENBLAS_NUM_THREADS=0` 

I get the following overusage:

![image](https://github.com/skyfielders/python-skyfield/assets/28128048/93ba2b17-32b9-4007-adad-43f7f59a8ebc)

If i change the openblas settings then its fixed:

`export OPENBLAS_NUM_THREADS=1` 

![image](https://github.com/skyfielders/python-skyfield/assets/28128048/c1742bde-ad41-4dd6-b6ed-e0d2d1a33eee)

Usage is extremely low as expeccted
